presto-orc/src/main/java/com/facebook/presto/orc/OrcWriter.java

```diff
@@ -533,19 +540,20 @@ private List<DataOutput> bufferStripeData(long stripeStartOffset, FlushReason fl
             return ImmutableList.of();
         }
 
-        List<DataOutput> outputData = new ArrayList<>();
         List<Stream> unencryptedStreams = new ArrayList<>(columnWriters.size() * 3);
         Multimap<Integer, Stream> encryptedStreams = ArrayListMultimap.create();
+        List<StreamDataOutput> indexStreams = new ArrayList<>(columnWriters.size());
```

Comment:
Previously we used List<DataOutput> for accumulating stream producers (StreamDataOutput) and a file footer producer (DataOutput). I switched it to using separate collections for streams to be able to pass all streams to the StreamSizeHelper as a List<StreamDataOutput>.